contributingGuides/STYLE.md

```diff
@@ -477,20 +477,63 @@ if (ref.current && 'getBoundingClientRect' in ref.current) {
 
- Use `'-1'` or `-1` when there is a possibility that the ID property of an Onyx value could be `null` or `undefined`.
+Use `CONST.DEFAULT_NUMBER_ID` when there is a possibility that the number ID property of an Onyx value could be `null` or `undefined`. **Do not default string IDs to any value unless absolutely necessary**, in case it's necessary use `CONST.DEFAULT_STRING_ID` instead.
 
 ``` ts
 // BAD
-const foo = report?.reportID ?? '';
-const bar = report?.reportID ?? '0';
+const accountID = report?.ownerAccountID ?? -1;
+const accountID = report?.ownerAccountID ?? 0;
+const reportID = report?.reportID ?? '-1';
 
-report ? report.reportID : '0';
-report ? report.reportID : '';
+// BAD
+report ? report.ownerAccountID : -1;
+
+// GOOD
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const reportID = report?.reportID;
 
 // GOOD
-const foo = report?.reportID ?? '-1';
+report ? report.ownerAccountID : CONST.DEFAULT_NUMBER_ID;
+```
+
+Here are some common cases you may face when fixing your code to remove the default values.
+
+#### **Case 1**: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
+
+```diff
+-Report.getNewerActions(newestActionCurrentReport?.reportID ?? '-1', newestActionCurrentReport?.reportActionID ?? '-1');
++Report.getNewerActions(newestActionCurrentReport?.reportID, newestActionCurrentReport?.reportActionID);
+```
+
+> error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'. Type 'undefined' is not assignable to type 'string'.
+
+We need to change `Report.getNewerActions()` arguments to allow `undefined`. By doing that we could add a condition that return early if one of the parameters are falsy, preventing the code (which is expecting defined IDs) from executing.
+
+```diff
+-function getNewerActions(reportID: string, reportActionID: string) {
++function getNewerActions(reportID: string | undefined, reportActionID: string | undefined) {
++    if (!reportID || !reportActionID) {
++        return;
++    }
+```
+
+#### **Case 2**: Type 'undefined' cannot be used as an index type.
+
+```diff
+function MoneyRequestView({report, shouldShowAnimatedBackground, readonly = false, updatedTransaction, isFromReviewDuplicates = false}: MoneyRequestViewProps) {
+-    const parentReportAction = parentReportActions?.[report?.parentReportActionID ?? '-1'];
++    const parentReportAction = parentReportActions?.[report?.parentReportActionID];
+```
+
+> error TS2538: Type 'undefined' cannot be used as an index type.
+
+This error is inside a component, so we can't just make conditions with early returns here. We can instead use `String(report?.parentReportActionID)` to try to convert the value to `string`. If the value is `undefined` the result string will be `'undefined'`, which will be used to find a record inside `parentReportActions` and, same as `-1`, would find nothing.
```

Comment:
I don't get why we are suggesting this. Why not default to `""`? We never want `"undefined"` to be used

Reply:
We are suggesting to remove the default value because `parentReportActionID` is a `string` and according to the [final proposal](https://github.com/Expensify/App/issues/50360#issuecomment-2432767978) we agreed to **not** default string IDs to any value, unless absolutely necessary.

> Update guidelines to require developers to default all number IDs to 0 and forbid defaulting string IDs unless absolutely necessary. Include examples about how to deal with common scenarios when removing the default values.

When using `String(report?.parentReportActionID)`, we will have `'undefined'` if `report?.parentReportActionID` is `undefined`, so we will try to access the `'undefined'` record of `parentReportActions`. Same as `'-1'` or even `''`, this record should never exist so it's safe to do it. 

However, I can change the diff example to make it clear that I was talking about Onyx usage, wdyt?


```diff
function MoneyRequestView({report, shouldShowAnimatedBackground, readonly = false, updatedTransaction, isFromReviewDuplicates = false}: MoneyRequestViewProps) {
    const [parentReportActions] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`);
-   const parentReportAction = parentReportActions?.[report?.parentReportActionID ?? '-1'];
+   const parentReportAction = parentReportActions?.[report?.parentReportActionID];
```